internal/core/src/common/ElementFilterIterator.cpp

```diff
@@ -0,0 +1,117 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ElementFilterIterator.h"
+
+#include "common/EasyAssert.h"
+#include "exec/expression/EvalCtx.h"
+#include "exec/expression/Expr.h"
+
+namespace milvus {
+
+ElementFilterIterator::ElementFilterIterator(
+    std::shared_ptr<VectorIterator> base_iterator,
+    exec::ExecContext* exec_context,
+    exec::ExprSet* expr_set)
+    : base_iterator_(std::move(base_iterator)),
+      exec_context_(exec_context),
+      expr_set_(expr_set),
+      exhausted_(false) {
+    AssertInfo(base_iterator_ != nullptr, "Base iterator cannot be null");
+    AssertInfo(exec_context_ != nullptr, "Exec context cannot be null");
+    AssertInfo(expr_set_ != nullptr, "ExprSet cannot be null");
+}
+
+bool
+ElementFilterIterator::HasNext() {
+    // If cache is empty and base iterator not exhausted, fetch more
+    if (filtered_buffer_.empty() && !exhausted_) {
+        FetchAndFilterBatch();
+    }
+    return !filtered_buffer_.empty();
+}
+
+std::optional<std::pair<int64_t, float>>
+ElementFilterIterator::Next() {
+    if (!HasNext()) {
+        return std::nullopt;
+    }
+
+    auto result = filtered_buffer_.front();
+    filtered_buffer_.pop_front();
+    return result;
+}
+
+void
+ElementFilterIterator::FetchAndFilterBatch() {
+    constexpr size_t kBatchSize = 1024;
+
+    // Step 1: Fetch a batch from base iterator (up to kBatchSize elements)
+    FixedVector<int32_t> element_ids;
```

Comment:
why fixed vector?

Reply:
seems now distances_buffer_ is using std vector but element_ids_buffer_ is still using fixed vector. 

not requesting to use std::vector, just want to know the reason behind that decision